Trigger all container builds on base image changes

danjoneslf opened the pull request:
At present only the base image is rebuilt when the base image
dockerfile is changed. The other containers are only changed
when their code changes which means that we have terraform version
drift across our pipelines.

danjoneslf committed:
Trigger all container builds on base image changes

At present only the base image is rebuilt when the base image
dockerfile is changed. The other containers are only changed
when their code changes which means that we have terraform version
drift across our pipelines.

danjoneslf committed:
Fix pipeline serialisation

Child containers should only be triggered after the parent "base"
container is rebuilt and pushed to "latest"

denizgenc reviewed (dismissed):
Nice and simple change, looks good

danjoneslf commented:
The runtime's erroring. Concourse is running out of disk space. Not sure if that's because of the parallelism or maybe not cleaning up in docker.

danjoneslf committed:
Put all the builds in serial.

When we ran them all in parallel we ran out of disk space on the
concourse worker. Looks like the resource use of multiple docker
builds in parallel is more than our concourse workers are setup
to support.

denizgenc reviewed (approved):
Looks good, hopefully this won't cause errors on Concourse now. Also thank you for changing the underscores to dashes with the latest commit
